### ISSUE TITLE
[proposal] Discontinue Debian package support

### ISSUE THREAD

**Issue by markaren** [open-simulation-platform/libcosim#707]
I would like to see that we discontinoue supporting building libcosim using Debian packages.
I.e. require the use of Conan. This way, the project is simplified, we avoid potential version issues and opens the door for non-ubuntu packages to be included. Furthermore, we could potentially remove all `FindXXX.cmake` files.

**Comment by eidekrist** [comment]
The OSP committee meeting sees no immediate issue in removing Debian package support.

**Comment by kyllingstad** [comment]
While working on the upgrade to Conan 2 ( #747), I found that it was actually no big deal to maintain support for building on Debian without Conan. As long as I switched from building directly on the GitHub runner – which is a bad idea anyway, because the build environment there is anything but clean – to building on Debian Stable, everything worked without a hitch. In fact, it was *easier* to get things to build there than with Conan.

The reason that it's easy to support both is that all of our dependencies now provide CMake package configuration files which are included in the Debian packages. And as of recently, the Conan packages for the same dependencies are set up such that the `CMakeDeps` generator creates package configuration files which are compatible with the upstream ones (same target names, etc.).

So I would suggest that we keep Debian support as long as it's easy to do so, but have a rather low threshold for dropping it. There are a few good things to be said for not being tightly bound to Conan too...
